distributed/worker.py

```diff
@@ -2437,10 +2437,7 @@ def release_key(self, key, cause=None, reason=None, report=True):
                 # don't boot keys that are in flight
                 # we don't know if they're already queued up for transit
                 # in a gather_dep callback
-                if not dependency.dependents and dependency.state in (
-                    "waiting",
-                    "fetch",
-                ):
```

Comment:
anybody knows why we were skipping this for specific states? cc @gforsyth

Reply:
We were skipping it for `flight` since there's a potential race condition if we remove the task and then the `gather_dep` callback fires

Reply:
Right, there is even a comment about this a few lines above 🤦